src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -354,42 +312,6 @@ internal virtual void OnSetNodeDeltaState(DeltaComputeStateEventArgs e)
                 annotations, presets, resolver, info, verboseLogging, isTestMode, linterManager)
         { Guid = guid; }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="HomeWorkspaceModel"/> class
-        /// by given information about it and specified item collections
-        /// </summary>
-        /// <param name="engine"><see cref="EngineController"/> object assosiated with this home workspace
-        /// to coordinate the interactions between some DesignScript sub components.</param>
-        /// <param name="scheduler"><see cref="DynamoScheduler"/> object to add tasks in queue to execute</param>
-        /// <param name="factory">Node factory to create nodes</param>
-        /// <param name="traceData">Preloaded trace data</param>
-        /// <param name="nodes">Node collection of the workspace</param>
-        /// <param name="notes">Note collection of the workspace</param>
-        /// <param name="annotations">Group collection of the workspace</param>
-        /// <param name="presets">Preset collection of the workspace</param>
-        /// <param name="resolver">ElementResolver responsible for resolving 
-        /// a partial class name to its fully resolved name</param>
-        /// <param name="info">Information for creating custom node workspace</param>
-        /// <param name="verboseLogging">Indicates if detailed descriptions should be logged</param>
-        /// <param name="isTestMode">Indicates if current code is running in tests</param>
-        [Obsolete("please use the version with linterManager parameter.")]
-        public HomeWorkspaceModel(EngineController engine, 
-            DynamoScheduler scheduler, 
-            NodeFactory factory,
-            IEnumerable<KeyValuePair<Guid, List<CallSite.RawTraceData>>> traceData, 
-            IEnumerable<NodeModel> nodes, 
-            IEnumerable<NoteModel> notes, 
-            IEnumerable<AnnotationModel> annotations,
-            IEnumerable<PresetModel> presets,
-            ElementResolver resolver,
-            WorkspaceInfo info, 
-            bool verboseLogging,
-            bool isTestMode)
-            : base(nodes, notes,annotations, info, factory,presets, resolver)
-        {
-            InitializeHomeWorkspace(engine, traceData, scheduler, info, verboseLogging, isTestMode);
-        }
-
```

Comment:
The remaining constructor seems to leave us in the same place of needing to make obsolete versions of these objects... I'm not sure what a good pattern is but perhaps thinking of options objects when we reach this number of dependencies.

Anyhow, maybe out of scope for 3.0

Reply:
Yep, out of scope.